src/ui/public/vis/editors/default/agg_params.js

```diff
@@ -106,36 +152,38 @@ uiModules
           };
 
           // build collection of agg params html
-          $scope.agg.type.params.forEach(function (param, i) {
-            let aggParam;
-            let fields;
-            if ($scope.agg.schema.hideCustomLabel && param.name === 'customLabel') {
-              return;
-            }
-            // if field param exists, compute allowed fields
-            if (param.name === 'field') {
-              fields = $aggParamEditorsScope.indexedFields;
-            } else if (param.type === 'field') {
-              fields = $aggParamEditorsScope[`${param.name}Options`] = param.getFieldOptions($scope.agg);
-            }
-
-            if (fields) {
-              const hasIndexedFields = fields.length > 0;
-              const isExtraParam = i > 0;
-              if (!hasIndexedFields && isExtraParam) { // don't draw the rest of the options if there are no indexed fields.
+          $scope.agg.type.params
+            .filter(param => !get($scope, ['editorConfig', param.name, 'hidden'], false))
```

Comment:
To make this easier to scan, can we assign this to a new variable or add a comment explaining what the intention is?

```js
const visibleParams =
  $scope.agg.type.params.filter(param => !get($scope, ['editorConfig', param.name, 'hidden'], false));
```